File: document-workspace-backend/src/main/java/com/pablodev/documentworkspace/dto/document/DocumentResponse.java
package com.pablodev.documentworkspace.dto.document;

import com.pablodev.documentworkspace.dto.extension.ExtensionResponse;
import com.pablodev.documentworkspace.dto.folder.FolderInfoResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@Builder
public class DocumentResponse {
    private Long id;
    private String filename;
    private ExtensionResponse extension;
    private Long length;
    private boolean locked;
    private Long version;
    private FolderInfoResponse folder;
    private LocalDateTime createdDate;
    private LocalDateTime lastModifiedDate;
}

File: document-workspace-frontend/src/app/core/model/document/document-response.ts
import {ExtensionResponse} from '../extension/extension-response';

export interface DocumentResponse {
  id: number;
  filename: string;
  length: number;
  extension: ExtensionResponse;
  createdDate: string;
  lastModifiedDate: string;
  locked: boolean;
  version: number;
}
